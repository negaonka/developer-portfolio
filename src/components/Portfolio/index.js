import React, { useState} from 'react';
import Project from '../Project';
import developer1 from  "../../assets/developer_images/developer1.jfif";
import developer2 from  "../../assets/developer_images/developer2.jfif";
import developer3 from  "../../assets/developer_images/developer3.jfif";
import developer4 from  "../../assets/developer_images/developer4.jfif";
import developer5 from  "../../assets/developer_images/developer5.jfif";
import developer6 from  "../../assets/developer_images/developer6.jfif";

function Portfolio(){

    const [developers] = useState([
        {
            image: developer1,
            application:"http://negaonka.github.io/photo-portfolio",
            github:"https://github.com/negaonka/photo-portfolio",
            name: "Photo Portfolio"
        },
        {
            image: developer2,
            application:"http://negaonka.github.io/photo-portfolio",
            github:"https://github.com/negaonka/photo-portfolio",
            name: "Photo Portfolio"
        },
        {
            image: developer3,
            application:"https://negaonka.github.io/hangman",
            github:'https://github.com/negaonka/hangman',
            name: "Technical LOL"
        },
        {
            image: developer4,
            application:"http://negaonka.github.io/SmartVolunteer",
            github:"https://github.com/negaonka/SmartVolunteer",
            name: "Smart Volunteer"
        },
        {
            image: developer5,
            application:"http://negaonka.github.io/photo-portfolio",
            github:"https://github.com/negaonka/photo-portfolio",
            name: "Photo Portfolio Test"
        },
        {
            image: developer6,
            application:"http://negaonka.github.io/SmartVolunteer",
            github:"https://github.com/negaonka/SmartVolunteer",
            name: "Smart Volunteer"
        }
      ]);

    return (
        <>
            <h1 data-testid="h1tag">Portfolio</h1>
            <div className="flex-row">
                {developers.map((developer) => (
                <Project developer = {developer}/>
            ))}
            </div>
      </>
    )

}

export default Portfolio;