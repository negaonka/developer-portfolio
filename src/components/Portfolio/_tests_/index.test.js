import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Portfolio from '..';

afterEach(cleanup);  //to clean all the memory data

describe('Portfolio component', () => {
    // baseline test
    it('renders', () => {
        render(<Portfolio />);
      });
   
    // snapshot test
    it('matches snapshot', () => {
        const { asFragment } = render(<Portfolio />);
        // assert value comparison
        expect(asFragment()).toMatchSnapshot();
      });

    })