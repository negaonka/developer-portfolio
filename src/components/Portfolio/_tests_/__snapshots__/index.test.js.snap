// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio component matches snapshot 1`] = `
<DocumentFragment>
  <h1
    data-testid="h1tag"
  >
    Portfolio
  </h1>
  <div
    class="flex-row"
  >
    <div>
      <div
        class="container"
      >
        <img
          alt="developer"
          height="300px"
          src="developer1.jfif"
          width="500px"
        />
        <div
          class="middle"
        >
          <a
            href="https://github.com/mojombo"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="github"
              height="40px"
              src="github.png"
              width="40px"
            />
          </a>
          <a
            href="https://api.github.com/users/mojombo"
            rel="noreferrer"
            target="_blank"
          >
            <p>
              Plantonic Game
            </p>
          </a>
        </div>
      </div>
    </div>
    <div>
      <div
        class="container"
      >
        <img
          alt="developer"
          height="300px"
          src="developer2.jfif"
          width="500px"
        />
        <div
          class="middle"
        >
          <a
            href="https://github.com/defunkt"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="github"
              height="40px"
              src="github.png"
              width="40px"
            />
          </a>
          <a
            href="https://api.github.com/users/defunkt"
            rel="noreferrer"
            target="_blank"
          >
            <p>
              Game Kaboolush
            </p>
          </a>
        </div>
      </div>
    </div>
    <div>
      <div
        class="container"
      >
        <img
          alt="developer"
          height="300px"
          src="developer3.jfif"
          width="500px"
        />
        <div
          class="middle"
        >
          <a
            href="https://github.com/pjhyett"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="github"
              height="40px"
              src="github.png"
              width="40px"
            />
          </a>
          <a
            href="https://api.github.com/users/pjhyett"
            rel="noreferrer"
            target="_blank"
          >
            <p>
              Technical LOL
            </p>
          </a>
        </div>
      </div>
    </div>
    <div>
      <div
        class="container"
      >
        <img
          alt="developer"
          height="300px"
          src="developer4.jfif"
          width="500px"
        />
        <div
          class="middle"
        >
          <a
            href="https://github.com/wycats"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="github"
              height="40px"
              src="github.png"
              width="40px"
            />
          </a>
          <a
            href="https://api.github.com/users/wycats"
            rel="noreferrer"
            target="_blank"
          >
            <p>
              Smart Kitten
            </p>
          </a>
        </div>
      </div>
    </div>
    <div>
      <div
        class="container"
      >
        <img
          alt="developer"
          height="300px"
          src="developer5.jfif"
          width="500px"
        />
        <div
          class="middle"
        >
          <a
            href="https://github.com/ezmobius"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="github"
              height="40px"
              src="github.png"
              width="40px"
            />
          </a>
          <a
            href="https://api.github.com/users/ezmobius"
            rel="noreferrer"
            target="_blank"
          >
            <p>
              Zombie Sculpture
            </p>
          </a>
        </div>
      </div>
    </div>
    <div>
      <div
        class="container"
      >
        <img
          alt="developer"
          height="300px"
          src="developer6.jfif"
          width="500px"
        />
        <div
          class="middle"
        >
          <a
            href="https://github.com/ivey"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="github"
              height="40px"
              src="github.png"
              width="40px"
            />
          </a>
          <a
            href="https://api.github.com/users/ivey"
            rel="noreferrer"
            target="_blank"
          >
            <p>
              Fitness Money
            </p>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
